Std/Data/Bool.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2023 F. G. Dorais. No rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: F. G. Dorais
+-/
+
+import Std.Tactic.Alias
+
+/-- Boolean exclusive or -/
+abbrev xor : Bool → Bool → Bool := bne
+
+namespace Bool
+
+/- Namespaced versions that can be used instead of prefixing `_root_` -/
+@[inherit_doc not] protected abbrev not := not
+@[inherit_doc or]  protected abbrev or  := or
+@[inherit_doc and] protected abbrev and := and
+@[inherit_doc xor] protected abbrev xor := xor
+
+instance (p : Bool → Prop) [inst : DecidablePred p] : Decidable (∀ x, p x) :=
+  match inst true, inst false with
+  | isFalse ht, _ => isFalse fun h => absurd (h _) ht
+  | _, isFalse hf => isFalse fun h => absurd (h _) hf
+  | isTrue ht, isTrue hf => isTrue fun | true => ht | false => hf
+
+instance (p : Bool → Prop) [inst : DecidablePred p] : Decidable (∃ x, p x) :=
+  match inst true, inst false with
+  | isTrue ht, _ => isTrue ⟨_, ht⟩
+  | _, isTrue hf => isTrue ⟨_, hf⟩
+  | isFalse ht, isFalse hf => isFalse fun | ⟨true, h⟩ => absurd h ht | ⟨false, h⟩ => absurd h hf
+
+instance : LE Bool := leOfOrd
+instance : LT Bool := ltOfOrd
+instance : Max Bool := maxOfLe
+instance : Min Bool := minOfLe
```

Comment:
These change the defeq of the boolean order to be (IMO) worse. Was this a deliberate change?

Reply:
There was a hope long ago that core would fixup `Ord` before this PR landed. Without that fixup, this is indeed a bad choice. My preferred versions are:
```lean
le x y := !x || y
lt x y := !x && y
max x y :=  x || y
min x y := x && y
```
These lead to the most efficient code in each case. I'm not sure if you really prefer your version for `le`? I will submit a small follow-up PR fixing this as soon as I have a chance.